### DISCUSSION TITLE
Unable to run postgres:12.17 in gh actions

### DISCUSSION THREAD

**Discussion by indiVar0508** [docker-library/postgres#1177] [General]
hi 

i was trying to use postgres:12.17 in gh actions, but not able to start service in my job i get below shown error while starting action job service container
![image](https://github.com/docker-library/postgres/assets/23294511/6d0217d8-519e-40ca-adce-c755bd611d9b)
I see that in local in docker the container restarts automatically but same is not happening for github service container.
![image](https://github.com/docker-library/postgres/assets/23294511/1ddbde70-fba6-4248-9418-b825fbe62e5c)

```
        options: >-
          --health-cmd "pg_isready -U $$POSTGRES_USER -h $$(hostname -i)"
          --health-interval 60s
          --health-timeout 30s
          --health-retries 8
```

This is the Health command i am using.

**Comment by indiVar0508** [comment]
if i remove the healthcheck command to be only "pg_isready" it works, but then i get error message mentioned in https://github.com/docker-library/postgres/issues/838

**Comment by tianon** [comment]
Instead of `$$(hostname -i)`, can you try hard-coded `127.0.0.1`?

**Reply by LaurentGoderre** [reply]
Doesn't omitting the host check itself?

**Reply by indiVar0508** [reply]
Yes thanks hardcoding 127.0.0.1 worked for me i also had to add `-d db_name`, 
my final command looked like `pg_isready -U user_name -d db_name -h 127.0.0.1 -p 5432`
i was just wondering isn't this wrong that health marks container healthy but is silently failing in the background?
like if i just do `pg_isready` container is marked healthy but i get a warning log in DB that `role doesn't exists`  like #838, even container should be marked unhealthy if command is wrong.

**Reply by tianon** [reply]
When you run this container the first time, it runs PostgreSQL twice -- once listening _only_ locally in the container (on the appropriate unix socket) for running `initdb` and initialization scripts, then restarts listening externally.  Thus, if you want to verify that the container is actually "ready" (fully initialized, all roles, etc), you have to connect via TCP.
